tenants-certified-attributes-updater/src/main/scala/it/pagopa/interop/tenantscertifiedattributesupdater/Dependencies.scala

```diff
@@ -0,0 +1,163 @@
+    Try(
+      new DefaultInteropTokenGenerator(
+        signerService(blockingEc),
+        new PrivateKeysKidHolder {
+          override val RSAPrivateKeyset: Set[KID] = ApplicationConfiguration.rsaKeysIdentifiers
+          override val ECPrivateKeyset: Set[KID]  = ApplicationConfiguration.ecKeysIdentifiers
+        }
+      )
+    ).toFuture
+
+  def createAction(
+    institutions: List[Institution],
+    tenants: List[PersistentTenant],
+    attributesIndex: Map[UUID, AttributeInfo]
+  ): TenantActions = {
+    val fromRegistry: Map[PersistentExternalId, AttributeInfo] =
+      institutions
+        .filter(_.id.nonEmpty)
+        .map { institution =>
+          PersistentExternalId(institution.origin, institution.id) -> AttributeInfo(
+            institution.origin,
+            institution.category
+          )
+        }
+        .toMap
+
+    val fromTenant: Map[PersistentExternalId, List[AttributeInfo]] =
+      tenants.map(tenant => tenant.externalId -> tenant.attributes.flatMap(attr => attributesIndex.get(attr.id))).toMap
+
+    val activations: List[InternalTenantSeed] =
+      fromRegistry
+        .filter(i => !fromTenant.contains(i._1))
+        .map { case (extId, attr) =>
+          InternalTenantSeed(ExternalId(extId.origin, extId.value), Seq(InternalAttributeSeed(attr.origin, attr.code)))
+        }
+        .toList
+
+    val revocations: Map[PersistentExternalId, List[AttributeInfo]] =
+      fromTenant.filter { case (extId, _) => !fromRegistry.contains(extId) }
+
+    TenantActions(activations, revocations)
+  }
+
+  def createAttributesIndex(attributes: Seq[PersistentAttribute]): Map[UUID, AttributeInfo] =
+    attributes
+      .filter(_.kind == Certified)
+      .mapFilter(attribute =>
+        attribute.origin.zip(attribute.code).map { case (origin, code) => attribute.id -> AttributeInfo(origin, code) }
+      )
+      .toMap
+
+  def retrieveAllInstitutions(
+    institutionsRetriever: (Int, Int) => Future[Institutions],
+    page: Int,
+    acc: List[Institution]
+  )(implicit
+    actorSystem: ActorSystem[_],
+    executionContext: ExecutionContext,
+    logger: LoggerTakingImplicit[ContextFieldsToLog]
+  ): Future[List[Institution]] = institutionsRetriever(page, maxLimit).flatMap(institutions =>
+    if (institutions.totalCount == acc.size) {
+      logger.info(s"Load completed size ${acc.size}")
+      Future.successful(acc)
+    } else retrieveAllInstitutions(institutionsRetriever, page + 1, acc ++ institutions.items)
+  )
+
+  def processActivations(tenantService: InternalTenantSeed => Future[Unit], list: List[List[InternalTenantSeed]])(
+    implicit
+    actorSystem: ActorSystem[_],
+    executionContext: ExecutionContext,
+    logger: LoggerTakingImplicit[ContextFieldsToLog]
+  ): Future[Unit] = list match {
+    case Nil       => Future.unit
+    case x :: Nil  => Future.traverse(x)(tenantService).map(_ => ())
```

Comment:
Since you already have `cats.implicits._` in scope `.map(_ =>())` can be both refactored using `.as(())` or `.void`